### PR TITLE
Show full module path in documentation

### PR DESCRIPTION
**Pull request by josh146**

### CODE REVIEW

**Comment by dime10** [comment]
Not exactly what I was expecting, but definitely an improvement:
Screenshot 2024-06-10 at 1 37 12 PM

For comparison how functions show up in PL:
Screenshot 2024-06-10 at 1 38 07 PM

Seems to be the opposite somehow.

Did you also want to the list of compiler pipelines for `print_compilation_stage` in this PR?

File: doc/conf.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ def __getattr__(cls, name):
 # Add any paths that contain templates here, relative to this directory.
 from pennylane_sphinx_theme import templates_dir
```
Comment (josh146) [inline doc/conf.py:204]:
```suggestion
```

**Comment by josh146** [comment]
@dime10 this should now work

**Review by dime10 — APPROVED** [review]
Results look good, no idea about the added templates 😆

**Pull request merged**